### PR TITLE
Improve IP restriction handling

### PR DESCRIPTION
**Pull request by robgruen**
This pull request introduces significant improvements to network security and HTTP request handling across the workflow engine and copilot plugin packages. The main focus is on centralizing and strengthening the detection and blocking of private or reserved network targets, refactoring HTTP GET logic for better security and maintainability, and updating dependencies to use shared utilities. It also enhances test coverage to ensure correct handling of edge-case IP addresses and network scenarios.

**Network security and HTTP request handling:**

* Refactored the `http.get` task in `builtinTasks.ts` to use new shared utilities for resolving and validating public IP addresses, improving detection of private/reserved addresses (including edge cases like decimal, octal, and hexadecimal IPv4, IPv4-mapped IPv6, Teredo, 6to4, NAT64, and unique-local IPv6). It now blocks credentialed URLs and enforces allowed/blocked host constraints more robustly. (`ts/examples/workflow/engine/src/builtinTasks.ts`)
* Replaced custom IP parsing and public/private network detection logic in `workspaceServer.ts` with centralized helpers from `@typeagent/common-utils/network`, ensuring consistent enforcement of network security across packages. (`ts/packages/copilot-plugin/src/mcp/workspaceServer.ts`) [[1]](diffhunk://#diff-49e590ac5f1e262cd4d008bef991d6dfdc48bc211462a42710cf52d097191b28L511-L602) [[2]](diffhunk://#diff-49e590ac5f1e262cd4d008bef991d6dfdc48bc211462a42710cf52d097191b28L611-R521) [[3]](diffhunk://#diff-49e590ac5f1e262cd4d008bef991d6dfdc48bc211462a42710cf52d097191b28L623-R533)

**Dependency and codebase updates:**

* Added `@typeagent/common-utils` as a dependency in both the workflow engine and copilot plugin packages to share network security logic. (`ts/examples/workflow/engine/package.json`, `ts/packages/copilot-plugin/package.json`, `ts/packages/utils/commonUtils/package.json`) [[1]](diffhunk://#diff-26e744bded0cdd5eb355db5b9746fa6690ef594af2e6bdce23b0aecd3dc13d56R33) [[2]](diffhunk://#diff-29dd0150224473d8d975ad248f9b38d81a945071ca3d83ba608fe0da41991cb5R36) [[3]](diffhunk://#diff-fcb8fcc7de3f89540e06c89943caf8e584e99a2ea8df439aa20f9beb9292da8fL18-R19)

**Testing improvements:**

* Expanded test cases in both the workflow engine and copilot plugin to cover a wide range of private/reserved network targets, including unusual address formats, ensuring that these are correctly blocked. (`ts/examples/workflow/engine/test/engine.spec.ts`, `ts/packages/copilot-plugin/test/workspaceServer.spec.ts`) [[1]](diffhunk://#diff-574edfff86a7a0b4f4a99294a6e06d28a2b4f29869b13ea94b566ce3dd070936L4871-R4915) [[2]](diffhunk://#diff-6ee799f1b33aae38fba4b6ea16cd5d4e80997a320a03691a46695bd55d110e4eR147-R163)
* Updated tests to reflect new error messages and validation logic, such as blocking credentialed HTTP/HTTPS URLs. (`ts/examples/workflow/engine/test/engine.spec.ts`)

**Internal utility improvements:**

* Exported the `readHttpResponseBody` helper for direct testing and use in other modules, facilitating better testability and code reuse. (`ts/examples/workflow/engine/src/builtinTasks.ts`, `ts/examples/workflow/engine/test/engine.spec.ts`)

These changes collectively harden the system against SSRF and similar attacks, ensure consistent network policy enforcement, and improve maintainability by consolidating network utilities.

### CODE REVIEW

File: ts/examples/workflow/engine/src/builtinTasks.ts [hunk 1]
```diff
@@ -498,6 +506,184 @@ export const stringSplit: TaskDefinition<
...
+    }
+
+    const request = url.protocol === "https:" ? httpsRequest : httpRequest;
+    return new Promise((resolveRequest, rejectRequest) => {
+        const req = request(
+            url,
+            {
+                headers,
+                ...createPinnedLookup(resolved),
+                signal,
```
Comment (jebrans) [inline ts/examples/workflow/engine/src/builtinTasks.ts:609]: Can we add a timeout here

**Review by jebrans — APPROVED** [review]

File: ts/examples/workflow/engine/src/builtinTasks.ts (hunk 1 above)
Comment (robgruen) [reply ts/examples/workflow/engine/src/builtinTasks.ts:609]: @copilot
